Stack/Opc.Ua.Core/Security/Certificates/CertificateValidator.cs

```diff
@@ -919,6 +938,54 @@ protected virtual async Task InternalValidate(X509Certificate2Collection certifi
                     target = new CertificateIdentifier(certificate);
                 }
 
+                foreach (X509ChainStatus chainStatus in chain.ChainStatus)
+                {
+                    switch (chainStatus.Status)
+                    {
+                        // status codes that are handled in CheckChainStatus
+                        case X509ChainStatusFlags.RevocationStatusUnknown:
+                        case X509ChainStatusFlags.Revoked:
+                        case X509ChainStatusFlags.NotValidForUsage:
+                        case X509ChainStatusFlags.OfflineRevocation:
+                        case X509ChainStatusFlags.InvalidBasicConstraints:
+                        case X509ChainStatusFlags.NotTimeValid:
+                        case X509ChainStatusFlags.NotTimeNested:
+                        case X509ChainStatusFlags.NoError:
+                            break;
+
+                        // by design, the trust root is not in the default store
+                        case X509ChainStatusFlags.UntrustedRoot:
+                            break;
```

Comment:
X509Chain won't check revocation if it doesn't trust the root (because it can't decide if it should trust the CRL/OCSP signatures).

If you know what your root should be already, add it to `chain.Policy.CustomTrustStore` and set `chain.Policy.CustomTrustMode` to `CustomRootTrust`.